### PR TITLE
Fix/moist heat capacity

### LINKED ISSUE #184: Possible typo in thermo.py under def c_pm()
Current : 
return constants.c_pd * (1.0 + (constants.c_pv / constants.c_pd + 1.0) * q)

Proposed modification :
return constants.c_pd * (1.0 + (constants.c_pv / constants.c_pd - 1.0) * q)

Reason:
When q(specific humidity) is 0, c_pm = c_pd but when q=1, c_pm should be equal to c_pv, which is not the case.

### PR DESCRIPTION
**Pull request by thabbott**
Closes #184

## Changes

Bugfix for #184 is implemented as discussed in the issue, and `thermo.c_pm` is now tested in `tests/units/test_thermo_sac`.

The bugfix has enough of a downstream effect on SAC calculations to cause failures for a small number of tests of pinned CoCiP output values, so I've updated the pinned values.

#### Breaking changes

- Bugfix will decrease computed values of moist heat capacity with non-zero specific humidity.

#### Fixes

- Fix moist heat capacity bug in `thermo.c_pm`.

#### Internals

- Update pinned CoCiP output values after moist heat capacity bugfix.

## Tests

- [x] QC test passes locally (`make test`)
- [x] CI tests pass

## Reviewer

> @mlshapiro

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -16,6 +17,7 @@
 ### Fixes
 
 - Use horizontal great circle distance to determine whether geodesic interpolation is used in `Flight.resample_and_fill`. This ensures geodesic interpolation is used between sufficiently distant waypoints even when one or both waypoints contain NaN altitude data.
+- Fix moist heat capacity bug in `thermo.c_pm`.
```
Comment (mlshapiro) [inline CHANGELOG.md:20]:
```suggestion
- Fix typo in moist heat capacity equation `thermo.c_pm`.
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -6,6 +6,7 @@
 ### Breaking changes
 
 - Geodesic interpolation is now used in `Flight.resample_and_fill` when the great circle distance between waypoints (rather than the total segment length including vertical displacement) exceeds a threshold. This may change the interpolation method used when resampling flight segments with lengths close to the geodesic interpolation threshold.
+- Bugfix will decrease computed values of moist heat capacity with non-zero specific humidity.
```
Comment (mlshapiro) [inline CHANGELOG.md:9]:
```suggestion
- Fixed typo in `thermo.c_pm`  will decrease computed values of moist heat capacity with non-zero specific humidity. This update results in a negligible (<<1%) change in contrail predictions by `ISSR`, `SAC`, `PCR, and `Cocip` models.
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -24,6 +26,7 @@
 - Update Makefile so `make ensure-era5-cached` uses default cache directory when run locally.
 - Bump pinned black and ruff versions.
 - Disable mypy type checking on `functools.wraps` in `support_arraylike` decorator to avoid error that appears starting on mypy 1.10.0.
+- Update pinned CoCiP output values after moist heat capacity bugfix.
```
Comment (mlshapiro) [inline CHANGELOG.md:29]:
```suggestion
- Update pinned `Cocip` test output values after moist heat capacity bugfix.
```

**Review by mlshapiro — APPROVED** [review]
Thanks! Feel free to merge

File: CHANGELOG.md (hunk 2 above)
Comment (thabbott) [reply CHANGELOG.md:9]: I'm a little hesitant to put a number on the downstream changes without actually quantifying them. The changes to values in pinned fixtures are all tiny, though, so I agree that the changes are likely to be negligible. How about
```
Fixed typo in `thermo.c_pm`  will decrease computed values of moist heat capacity with non-zero specific humidity. We expect the downstream impact on contrail predictions by `ISSR`, `SAC`, `PCR, and `Cocip` models to be negligible.
```
instead?

Comment (mlshapiro) [reply CHANGELOG.md:9]: I agree - and probably best to characterize as `minimal` rather than `negligble`. Thanks!

Comment (thabbott) [reply CHANGELOG.md]: https://github.com/contrailcirrus/pycontrails/pull/185/commits/d21451bc7c60f32b9e717cb22ebc1de455254ba3

**Pull request merged**